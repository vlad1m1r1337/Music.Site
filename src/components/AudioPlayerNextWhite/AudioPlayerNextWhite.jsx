import * as SAudio from "../AudioPlayer/AudioPlayer.styles";

export function AudioPlayerNextWhite({NotImplemented}) {
    return (
        <SAudio.PlayerBtnNext onClick={NotImplemented}>
            <SAudio.PlayerBtnNextSvg alt="next">
                <use xlinkHref="/img/icon/sprite.svg#icon-next-white"></use>
            </SAudio.PlayerBtnNextSvg>
        </SAudio.PlayerBtnNext>
    )
}
