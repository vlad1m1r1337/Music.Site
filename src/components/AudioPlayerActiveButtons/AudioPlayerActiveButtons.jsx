import {AudioPlayerButtonPrev} from "../AudioPlayerButtonPrev/AudioPlayerButtonPrev";
import AudioPlayerPlayButton from "../AudioPlayerPlayButton/AudioPlayerPlayButton";
import {AudioPlayerNextBlack} from "../AudioPlayerNextBlack/AudioPlayerNextBlack";
import {AudioPlayerButtonRepeatBlack} from "../AudioPlayerButtonRepeatBlack/AudioPlayerButtonRepeatBlack";
import {AudioPlayerBtnShuffle} from "../AudioPlayerBtnShuffle/AudioPlayerBtnShuffle";
import {AudioPlayerButtonPrevWhite} from "../AudioPlayerButtonPrevWhite/AudioPlayerButtonPrevWhite";
import {AudioPlayerPlayButtonWhite} from "../AudioPlayerPlayButtonWhite/AudioPlayerPlayButtonWhite";
import {AudioPlayerNextWhite} from "../AudioPlayerNextWhite/AudioPlayerNextWhite";
import {AudioPlayerButtonRepeatWhite} from "../AudioPlayerButtonRepeatWhite/AudioPlayerButtonRepeatWhite";
import styled from "styled-components";
import {useThemeContext} from "../../contexts/color_theme";
import {useDispatch} from "react-redux";
import {increment} from "../../store";
import {decrement} from "../../store";
import {shuffle_handler} from "../../store";
import {useState} from "react";

export const PlayerControls = styled.div`
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0 27px 0 31px;
`

export const AudioPlayerActiveButtons = ({audioRef}) => {
    const [repeat, setRepeat] = useState(true);

    const {theme} = useThemeContext();

    const [shuffle, setShuffle] = useState(false)

    const dispatch = useDispatch();

    const nextTrack = () => {
        if (shuffle) {
            dispatch(shuffle_handler())
        }
        else {
            dispatch(increment())
        }
        // shuffle ? dispatch(shuffle_handler()) : dispatch(increment());
    }
    const prevTrack = () => {
        if (shuffle) {
            console.log("some_prev")
            dispatch(shuffle_handler())
        }
        else {
            dispatch(decrement())
        }
    }
    function cycleExec() {
        setRepeat(!repeat);
        repeat ? audioRef.current.loop=true : audioRef.current.loop=false;
    }
    return (
        <PlayerControls>
            {theme.theme === "black" ? (
                <>
                    <AudioPlayerButtonPrev prevTrack={prevTrack}/>
                    <AudioPlayerPlayButton audioRef={audioRef}/>
                    <AudioPlayerNextBlack nextTrack={nextTrack}/>
                    <AudioPlayerButtonRepeatBlack repeat={repeat} cycleExec={cycleExec}/>
                    <AudioPlayerBtnShuffle $theme={theme} setShuffle={setShuffle} shuffle={shuffle}/>

                </>
            ) : (
                <>
                    <AudioPlayerButtonPrevWhite prevTrack={prevTrack}/>
                    <AudioPlayerPlayButtonWhite audioRef={audioRef}/>
                    <AudioPlayerNextWhite nextTrack={nextTrack}/>
                    <AudioPlayerButtonRepeatWhite repeat={repeat} cycleExec={cycleExec}/>
                    <AudioPlayerBtnShuffle $theme={theme} setShuffle={setShuffle} shuffle={shuffle}/>
                </>
            )}
        </PlayerControls>
    )
}