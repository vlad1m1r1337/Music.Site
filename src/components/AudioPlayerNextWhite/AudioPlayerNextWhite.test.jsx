import {AudioPlayerNextWhite} from "./AudioPlayerNextWhite";
import {render} from "@testing-library/react";
import {nextTrack} from "../AudioPlayerActiveButtons/AudioPlayerActiveButtons";

describe("<AudioPlayerNextWhite/>", () => {
    it("should render successfully and match snapshot", () => {
        const { container } = render(<AudioPlayerNextWhite nextTrack={nextTrack} />);
        expect(container).toMatchSnapshot();
    })
});