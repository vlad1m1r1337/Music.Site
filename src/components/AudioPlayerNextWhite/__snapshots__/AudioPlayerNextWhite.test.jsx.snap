// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioPlayerNextWhite/> should render successfully and match snapshot 1`] = `
<div>
  <div
    class="sc-jsEeA-d kDKjQU"
  >
    <svg
      alt="next"
      class="sc-kFCrIq iLjXHm"
    >
      <use
        xlink:href="/img/icon/sprite.svg#icon-next-white"
      />
    </svg>
  </div>
</div>
`;
